### ISSUE TITLE
Deep relation when querying

### ISSUE THREAD

**Issue by JCWolf** [typeorm/typeorm#2125]
**Issue type:**

[x] question
[ ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

It is possible to pass a deep relation when querying? For example, I have a `User`, a `User` can have many `cars` and a `Car` has one `Make`. When I query the `User` I want to get his `cars` and each `Car`'s `Make`. I don't want to use Eager or Lazy loading. I'm thinking something like:

```ts
$users = User.find({relations : ['cars.make']})
```

If no, are there any plans for implementing it in the near future?
As an example, I saw this behavior in the [PHP Laravel Framework](https://stackoverflow.com/questions/30231862/laravel-eloquent-has-with-wherehas-what-do-they-mean?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

Thank you!

**Comment by pleerock** [comment]
yes its supported as [documentation says](http://typeorm.io/#/find-options/basic-options), did you try it?

**Comment by JCWolf** [comment]
Oh, you are right,, I have to explicitly write every relation. They are not recursively deduced from one parameter. I missed this in the docs. I would be nice to have the relations loaded recursively automatically.

Sorry for my lack of documentation.

Thank you!

**Comment by pleerock** [comment]
>  I would be nice to have the relations loaded recursively automatically.

as you already told its a recursive and can load the whole database for you which may not be expected.

**Comment by JCWolf** [comment]
I was referring to the fact that I have to explicitly write every relation like so:
```ts
$users = User.find({relations : ['cars','cars.make']})
```
But it would be nice to write it only once like so:
```ts
$users = User.find({relations : ['cars.make']})
```
It's kinda redundant to write it twice. If I want to remove the `cars` relation I have to remove any other relation linked with the `car` too.

**Comment by pleerock** [comment]
> User.find({relations: ['client.cars.version.model.make']});

hope you don't have such relation nesting in a real life 😨 

Yeah its redundant, but more explicit, so Im not sure about this change...

**Comment by JCWolf** [comment]
Ok. This is more clear now. Closing the issue.

**Comment by pleerock** [comment]
Thank you for the suggestion anyway. In 0.3.0 Im [planning](https://github.com/typeorm/typeorm/issues/2100) to refactor current way of using find options so I think this won't work this way anyway.

**Comment by RDeluxe** [comment]
@pleerock I'm puzzled, shouldn't
```ts
$users = User.find({relations : ['cars.make']})
```
works if `cars` is marked as `eager: true` ? It does not in our case

**Comment by pleerock** [comment]
Looks like a bug, but why do you put it into relations if you already eager it?

**Comment by pleerock** [comment]
btw you may want to checkout `@next` which find options where completely changed and should not have this issue

**Comment by RDeluxe** [comment]
I'm actually on @next, sorry I forgot to specify that. But I stuck to 0.2.6 for now, there was some bug and breaking changed in 0.3.0. I might try again .
Did you implement #2100 in @next ?

**Comment by pleerock** [comment]
yes I did

**Comment by Matfork** [comment]
Is there a similar workaround for deep joining when using the `join` property instead of the `realtion` one?

**Comment by lianghx-319** [comment]
so can query builder load the deep relations?

**Comment by dianjuar** [comment]
I need to copy an entity of a model that has several relations and is constantly changing so I can not afford put by hand all the relations.

I need a deep query param to be able to get in deep all the relations

**Comment by Chetan11-dev** [comment]
Thank you you solved my question. I am really greatful to you. You are doing great work without TypeOrm I cannot imagine fiddling with databases. God give you Long Life
